tfconfig/load_hcl.go

```diff
@@ -22,10 +22,20 @@ func loadModule(fs FS, dir string) (*Module, Diagnostics) {
 	for _, filename := range primaryPaths {
 		var file *hcl.File
 		var fileDiags hcl.Diagnostics
+
+		b, err := fs.ReadFile(filename)
+		if err != nil {
+			diags = append(diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Failed to read file",
+				Detail:   fmt.Sprintf("The configuration file %q could not be read.", filename),
+			})
```

Comment:
This error matches the one from `ParseHCLFile`: https://github.com/hashicorp/hcl/blob/350d663f3c09e5a6d72ca11cb3250c1ce395bc8f/hclparse/parser.go#L70-L79

There is a small drift from `ParseJSONFile` in that `json.ParseFile` first opens the file and reports early diagnostic from there. This difference seems insignificant to me, but I'm open to retrofitting the extra `Open` call with that extra dedicated diagnostic here if deemed necessary.